### PR TITLE
GC: Refactor GC tombstone configs initialization to be in one place in garbage collector

### PR DESCRIPTION
**Pull request by agarwal-navin**
This PR moves the gc tomsbtone configs initalization into one place in garbage collector configs. Currently, the same configs are initialized in DataStores, DataStoreContext and GarbageCollector making it possible for them to get out of sync. Also, changing it requires changing them everywhere making changes more complex and prone to errors.

### CODE REVIEW

File: packages/runtime/container-runtime/src/gc/garbageCollection.ts [hunk 1]
```diff
@@ -113,6 +113,19 @@ export class GarbageCollector implements IGarbageCollector {
...
 
+	/** If false, loading or using a Tombstoned object should merely log, not fail */
+	public get gcTombstoneEnforcementAllowed(): boolean {
+		return this.configs.gcTombstoneEnforcementAllowed;
+	}
+	/** If true, throw an error when a tombstone data store is retrieved */
+	public get gcThrowOnTombstoneLoad(): boolean {
+		return this.configs.gcThrowOnTombstoneLoad;
+	}
+	/** If true, throw an error when a tombstone data store is used. */
```
Comment (markfields) [inline packages/runtime/container-runtime/src/gc/garbageCollection.ts:124]:
```suggestion
	/** If true, throw an error when a tombstone data store is used (changed or revived). */
```

Right?

File: packages/runtime/container-runtime/src/gc/gcConfigs.ts [hunk 2]
```diff
@@ -173,10 +188,13 @@ export function generateGCConfigs(
 		sessionExpiryTimeoutMs,
 		sweepTimeoutMs,
 		inactiveTimeoutMs,
-		throwOnInactiveLoad,
 		persistedGcFeatureMatrix,
 		gcVersionInBaseSnapshot,
 		gcVersionInEffect,
+		gcThrowOnInactiveLoad,
```
Comment (markfields) [inline packages/runtime/container-runtime/src/gc/gcConfigs.ts:194]: Why preface all these gc configs with `gc`?

File: packages/runtime/container-runtime/src/dataStoreContext.ts [hunk 3]
```diff
@@ -326,10 +326,7 @@ export abstract class FluidDataStoreContext
 		);
 
 		// Tombstone should only throw when the feature flag is enabled and the client isn't a summarizer
```
Comment (markfields) [inline packages/runtime/container-runtime/src/dataStoreContext.ts:328]: You can/should remove this comment

File: packages/runtime/container-runtime/src/gc/gcConfigs.ts [hunk 4]
```diff
@@ -24,8 +24,10 @@ import {
 	runSessionExpiryKey,
 	runSweepKey,
 	stableGCVersion,
+	throwOnTombstoneLoadKey,
+	throwOnTombstoneUsageKey,
 } from "./gcDefinitions";
-import { getGCVersion, shouldAllowGcSweep } from "./gcHelpers";
+import { getGCVersion, shouldAllowGcSweep, shouldAllowGcTombstoneEnforcement } from "./gcHelpers";
```
Comment (markfields) [inline packages/runtime/container-runtime/src/gc/gcConfigs.ts:30]: You can probably remove `shouldAllowGcTombstoneEnforcement` from `gc/index.ts`.  Maybe the two `Key`s above too?

**Review by markfields — APPROVED** [review]

File: packages/runtime/container-runtime/src/gc/garbageCollection.ts (hunk 1 above)
Comment (agarwal-navin) [reply packages/runtime/container-runtime/src/gc/garbageCollection.ts:124]: Not really. We don't throw on revived today.

File: packages/runtime/container-runtime/src/gc/gcConfigs.ts (hunk 2 above)
Comment (agarwal-navin) [reply packages/runtime/container-runtime/src/gc/gcConfigs.ts:194]: Initially, I was returning these directly from garbage collector to container runtime. I did not rename them when I changed them to getters.

File: packages/runtime/container-runtime/src/gc/gcConfigs.ts (hunk 4 above)
Comment (agarwal-navin) [reply packages/runtime/container-runtime/src/gc/gcConfigs.ts:30]: Good catch. On removing that, I realized BlobManager has the logic to initialize `tombstoneFailOnLoad` too.

**Pull request merged**